### DISCUSSION TITLE
Android (Java) Target Not Working

### DISCUSSION THREAD

**Discussion by RuiCuco** [Kode/Kha#1459] [General]
It has the item <Haxe 3.2.0> in the dependencies list (Android Studio complains that it is invalid). Also, khamake says:

node C:\Developer\Kha\make.js android 

Using Kha (b90792d0) from C:\Developer\Kha
Creating Kha project.
...
Using Kinc (06da8e4d) from C:\Developer\Kha\Kinc
kfile found.
Creating Android project files.
Platform is set to Android, but android toolchain not found.
Please set the ANDROID_NDK environment variable if you need a compile_commands.json for your IDE.
Done.
Done.

Android NDK should not be necessary for this one, I think.

**Comment by RobDangerous** [comment]
The Android (Java) target has been removed a long while ago, what you are using is the Android (C++) target - which requires the NDK.

**Comment by RuiCuco** [comment]
The target above is called just 'Android' (I assumed it was Java). I tried 'Android (Native)' without success, because it prints an error message:

Executing task: node C:\Developer\Kha\make.js android-native 

Using Kha (b90792d0) from C:\Developer\Kha
Creating Kha project.
Unknown platform: android-native (baseTarget=$android-native)

So, both Android targets don't work (see the dependencies error above, right in the first line). Which should I use?

**Comment by RobDangerous** [comment]
There is only one Android target nowadays. And it needs the NDK.

**Comment by RuiCuco** [comment]
I tried the Android target, only Android Studio says that the item <Haxe 3.2.0> in the module dependencies list is invalid, whatever that means. Can you help?

**Comment by RobDangerous** [comment]
Oh, I have an idea about that - did you maybe open the ProjectName-android-intellij directory instead of the android-build/ProjectName directory?

**Comment by RuiCuco** [comment]
Yes, that did it. Thanks! Now I've got a problem. When I say:

	c.drawTexture(x1, y1, texItem1);
	c.drawTexture(x2, y2, texItem2);
	c.drawTexture(x3, y3, texItem3);

only the last texture (texItem3) is drawn. The others are not drawn. If I reorder the items, only the last item in the list is drawn. Each drawTexture is basically:

	g4.setMatrix(cameraMatrixVar, cameraMatrix);
	g4.setMatrix(objectMatrixVar, objectMatrix);
	g4.setVector4(objectMaterialVar, Vector.initFromReals(material.r, material.g, material.b, material.a));
	g4.setTexture(objectTextureVar, texture.image);
	g4.setVertexBuffer(attrs.vertexBuffer);
	g4.setIndexBuffer(attrs.indexBuffer);
	g4.drawIndexedVertices();

The code works in HTML5 Debug, but not in Android. Can you tell me what is happening?

**Comment by RobDangerous** [comment]
Don't know. Linux and Android switched to Vulkan by default not too long ago, might be a bug with that. Can you open an issue and provide a small sample-project?
